### DISCUSSION TITLE
ios: supporting XCFrameworks

### DISCUSSION THREAD

**Discussion by Klug76** [airsdk/Adobe-Runtime-Support#1776] [Ideas]
It would be nice to support the XCFrameworks when building ane and ipa packages. [https://developer.apple.com/documentation/swift_packages/distributing_binary_frameworks_as_swift_packages](https://developer.apple.com/documentation/swift_packages/distributing_binary_frameworks_as_swift_packages)

**Comment by marchbold** [comment]
Not too much of a problem currently as we can use the xcode tools to split it up easily but wouldn't be a bad addition for the future 👍

**Reply by AleDaDa** [reply]
Can you tell me How can I split the xcframework to framework ?

**Reply by marchbold** [reply]
xcframeworks are just folders with the libs / frameworks for each platform contained within subdirectories. Just pull the libs / frameworks from the platform subdirectory you need.

**Reply by AleDaDa** [reply]
Thx for reply. 
I have tried to do so. But while compile ipa, I meet Err:
ld: warning: ignoring file /var/folders/1v/xdt6qc5x6rd_r5nkf_fskqgr0000gn/T/98a5866e-383d-4443-a3da-f0a01cb20baa/Frameworks/BURelyFoundation.framework/BURelyFoundation, building for iOS-arm64 but attempting to link with file built for unknown-unsupported file format ( 0x56 0x65 0x72 0x73 0x69 0x6F 0x6E 0x73 0x2F 0x43 0x75 0x72 0x72 0x65 0x6E 0x74 )

**Reply by marchbold** [reply]
I'd guess you picked the wrong platform?

**Reply by AleDaDa** [reply]
I have tried the "xxx.xcframework/ios-arm64_armv7/xxx.framework" and "xxx.xcframework/ios-arm64_i386_x86_64-simulator/xxx.framework".
All return the same err.
And I compile the "frameWork" folder under the xxx.xcframework with older version frameWork folder. It seems missing a  "info.plist" ?

**Reply by AleDaDa** [reply]
I think I got it.
I should copy "xx.framework/Versions/A/*" to replace the files in "xx.framework/".

**Reply by marchbold** [reply]
Normally they are symlinked to the correct location, when you are copying it are you preserving the symlinks etc?

**Reply by AleDaDa** [reply]
Yes , I delete the link files.

**Reply by marchbold** [reply]
Do not delete them, you are corrupting the framework if you do that.

**Reply by AleDaDa** [reply]
ok  thank you for your help
